lib/pbench/test/unit/server/query_apis/test_datasets_list.py

```diff
@@ -156,41 +131,58 @@ def test_query(self, client, server_config, query_helper, user_ok, find_template
         }
 
         index = self.build_index(server_config, ("2020-08", "2020-09", "2020-10"))
-        response = query_helper(json, index, 200, server_config, json=response_payload)
-        res_json = response.json
-        assert isinstance(res_json, list)
-        assert len(res_json) == 1
-        assert (
-            res_json[0]["key"]
-            == "fio_rhel8_kvm_perf43_preallocfull_nvme_run4_iothread_isolcpus_2020.04.29T12.49.13"
+        response = query_api(
+            json, index, HTTPStatus.OK, server_config, json=response_payload
         )
-        assert res_json[0]["@metadata.controller_dir"] == "dhcp31-187.example.com"
+        res_json = response.json
+        assert isinstance(res_json, dict)
+        assert len(res_json.keys()) == 1
+        data = res_json["dhcp31-187.example.com"]
+        assert isinstance(data, list)
+        assert len(data) == 1
+        assert data[0]["@metadata.controller_dir"] == "dhcp31-187.example.com"
         assert (
-            res_json[0]["run.config"]
+            data[0]["config"]
             == "rhel8_kvm_perf43_preallocfull_nvme_run4_iothread_isolcpus"
         )
-        assert res_json[0]["run.controller"] == "dhcp31-187.example.com"
+        assert data[0]["controller"] == "dhcp31-187.example.com"
         assert (
-            res_json[0]["run.name"]
+            data[0]["result"]
             == "fio_rhel8_kvm_perf43_preallocfull_nvme_run4_iothread_isolcpus_2020.04.29T12.49.13"
         )
-        assert res_json[0]["run.start"] == "2020-04-29T12:49:13.560620"
-        assert res_json[0]["run.end"] == "2020-04-29T13:30:04.918704"
-        assert res_json[0]["startUnixTimestamp"] == 1588167004918
-        assert res_json[0]["id"] == "12fb1e952fd826727810868c9327254f"
+        assert data[0]["start"] == "2020-04-29T12:49:13.560620"
+        assert data[0]["end"] == "2020-04-29T13:30:04.918704"
+        assert data[0]["id"] == "12fb1e952fd826727810868c9327254f"
 
     @pytest.mark.parametrize(
         "exceptions",
         (
-            {"exception": requests.exceptions.HTTPError, "status": 502},
-            {"exception": requests.exceptions.ConnectionError, "status": 502},
-            {"exception": requests.exceptions.Timeout, "status": 504},
-            {"exception": requests.exceptions.InvalidURL, "status": 500},
-            {"exception": Exception, "status": 500},
+            {
+                "exception": make_http_exception(HTTPStatus.BAD_REQUEST),
+                "status": HTTPStatus.BAD_GATEWAY,
+            },
+            {
+                "exception": requests.exceptions.ConnectionError,
+                "status": HTTPStatus.BAD_GATEWAY,
+            },
+            {
+                "exception": requests.exceptions.Timeout,
+                "status": HTTPStatus.GATEWAY_TIMEOUT,
+            },
```

Comment:
Do we really want to report these as "gateway" issues?  In these cases, the server is a client, not a proxy, right?  So, won't this confuse things if, for instance, there is an actual proxy between the actual client and the Pbench server?

Reply:
That's an issue completely separate from this PR, since it's existing code. The idea at the time was that we're reporting an external failure in a subsystem for which we're essentially acting as a proxy; but that really applies better to the pass-through Elasticsearch API than for the real server queries where the backend DB should be invisble.

Again, though, I'd rather not dilute the focus of this PR by dealing with those issues here.

Reply:
Tracked in #2327.  (GitHub makes it _very_ easy to create issues from PR comments!)